data/tekton-pipelines/okd/windows10-installer.yaml

```diff
@@ -0,0 +1,470 @@
+            <SynchronousCommand wcm:action="add">
+              <CommandLine>reg add "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" /v AutoLogonCount /t REG_DWORD /d 0 /f</CommandLine>
+              <RequiresUserInput>false</RequiresUserInput>
+              <Order>1</Order>
+              <Description>Set AutoLogonCount to 0</Description>
+            </SynchronousCommand>
+          </FirstLogonCommands>
+        </component>
+      </settings>
+    </unattend>
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: windows10-installer
+spec:
+  params:
+    - name: winImageDownloadURL
+      description: Download URL to Windows 10 installation ISO (English United States x64 version is needed).
+      type: string
+    - name: autounattendConfigMapName
+      description: Name of the ConfigMap containing the sysprep configuration files (autounattend.xml, etc.).
+      type: string
+      default: windows10-autounattend
+    - name: virtioContainerDiskName
+      description: Reference to the containerdisk containing the virtio-win drivers ISO.
+      type: string
+      default: quay.io/kubevirt/virtio-container-disk@sha256:3e88b9bde8792ba21d6cdaad70c79d5e9b339ef676402e932c6fcc2bda100d04
+    - name: sourceTemplateName
+      description: Name of the Template which is used to create the installer and golden Template.
+      type: string
+      default: windows10-desktop-large
+    - name: sourceTemplateNamespace
+      description: Namespace of the Template which is used to create the installer and golden Template.
+      type: string
+      default: openshift
+    - name: installerTemplateName
+      description: Name of the installer Template which is created. A VM created from this template is used to install Windows 10.
+      type: string
+      default: windows10-desktop-large-installer
+    - name: allowReplaceInstallerTemplate
+      description: Allow to replace an already existing installer Template.
+      type: string
+      default: "false"
+    - name: goldenTemplateName
+      description: Name of the golden Template which is created. Pre-installed Windows VMs can be created from this template.
+      type: string
+      default: windows10-desktop-large-golden
+    - name: allowReplaceGoldenTemplate
+      description: Allow to replace an already existing golden Template.
+      type: string
+      default: "false"
+  tasks:
+    - name: create-base-dv
+      params:
+        - name: manifest
+          value: |
+            apiVersion: cdi.kubevirt.io/v1beta1
+            kind: DataVolume
+            metadata:
+              generateName: windows10-base-
+            spec:
+              storage:
+                resources:
+                  requests:
+                    storage: 20Gi
+              source:
+                blank: {}
+      timeout: 10m
+      taskRef:
+        kind: ClusterTask
+        name: create-datavolume-from-manifest
+    - name: copy-template-installer
```

Comment:
Would be nice if we could delete this temporary template after it is not required anymore.
If we are not able to delete it, the pipeline must not fail, if the template already exists, e.g. because the pipeline is executed a second time.

Reply:
I have overseen the replace parameter, which is good enough.